### DISCUSSION TITLE
🌹 ROSE Bloom Grants • Build an app that primarily uses Parcel SDK

### DISCUSSION THREAD

**Discussion by jcueto** [oasisprotocol/community-forum#6] [Grants Program]
![image](https://user-images.githubusercontent.com/5066732/106105783-6276ad00-60f9-11eb-8a08-6b07d598e61e.png)
-----
### Apply here: https://oasisfoundation.typeform.com/to/HtYql2aN
-----

_Note: This discussion thread serves as a hub for grant applications related to this project category. You'll see project proposals that have been submitted and reviewed by the Oasis Foundation team. You'll see proposals that have been marked as "ACCEPTED" and "TURNED DOWN," which will give you a better sense of what we're looking for in a grant application. Community members are encouraged to upvote projects, provide feedback, ask questions, and offer suggestions on each project shared on this thread._ 

-----

The ROSE Bloom Grants Program is an entirely new grants program designed to accelerate the growth of the Oasis Network and build valuable new ecosystem tools. Apply for a grant today to bring your project to life!

For this specific project category, we’re looking for developers who want to build new applications that primarily leverage the Parcel SDK. Parcel SDK is a privacy-first data governance SDK built by Oasis Labs to isolate and protect your most sensitive data without sacrificing the ability to use it. You can learn more about Parcel’s features here: https://www.oasislabs.com/parcelsdk 

This project category is pretty open-ended. To qualify for a grant, an application simply has to focus on leveraging the Parcel SDK. You can use it to privately store your application’s data, manage access permissions, and more. You can build a responsible data app, a privacy-focused DeFi app, or any app of your liking. For more ideas on what you can build with Parcel, check out the “HELPFUL DOCUMENTATION” section of this post. 

-----

### HOW TO APPLY

Applying for a grant is easy. Just fill out the [application form](https://oasisfoundation.typeform.com/to/HtYql2aN) with the relevant information we are requesting, and we’ll notify you via email regarding the status of your application. 

The more specific and clear you are in your application, the better our team will be able to understand your project proposal.

-----

### GRANT AMOUNT

Each proposal will be eligible to receive anywhere from **$5,000 to $50,000 USD** given out in **ROSE tokens**. Please specify the grant amount you are requesting in your application. The grant amount requested should match the scope of work for each project. 

-----

### PROJECT MILESTONES

We ask that you provide specific project milestones as part of your application. Milestones should be decided based on the type of project you are developing and the scope of work for the project. In general, the types of things you can list as project milestones include specific development steps (e.g. initial deployment, debugging, obtaining first user, etc.), specific components of the application (e.g. sign-in flow, front-end user interface, database config, etc.), as well as key success metrics (e.g. app installs, number of transactions, number of users, etc.).

Depending on the size of your grant, the Oasis Foundation may disburse your grant in separate parts. Smaller grants may be disbursed in a single transaction. Larger grants may be disbursed in separate transactions at separate times, based on evidence of your continued progress building your project and meeting the milestones we agreed upon. 

Please note that we expect a commitment of at least around 3 months, including project development, launch, maintenance, etc. If you want to continue working on your project or developing extensions for it beyond this initial commitment, you can apply for additional grants in the future to fund your continued work. 

-----

### SELECTION CRITERIA

The Oasis Foundation team will assess your application based on the following scoring categories: 
- Strength of the team, based on past projects, resume, GitHub, etc. - 1 (Very Weak) to 5 points (Very Strong) 
- Written quality of project proposal, based on details described, content, etc. - 1 (Very Low Quality) to 5 points (Very High Quality)
- Tech & product quality of project, based on technical implementation details and product vision, design overview, etc. - 1 (Very Low Quality) to 5 points (Very High Quality)
- Relevance to the success of the network, based on whether this is a product that could help us grow, an app that is in-demand among our community members, etc.- 1 (Not very relevant at all) to 5 points (Very Relevant)

Community interest and excitement generated will be a factor considered in the selection process, as an input into the “Relevance to the success of the network” scoring category. We’ll post project overviews on the Oasis GitHub discussion board to allow Oasis community members to share feedback, ask questions, and offer suggestions for each project. 

-----

### APPLICATION TIMELINE

Grant applications will be reviewed on an ongoing, rolling basis. In general, the application process will follow the timeline below: 
- Apply for a grant via our application form: https://oasisfoundation.typeform.com/to/HtYql2aN 
- Allow 1 to 2 weeks for the Oasis team to do a preliminary review of your application
- If your project idea looks promising, the Oasis team will post your project overview on the Oasis GitHub discussion board and notify you via email (applicants that are turned down at this preliminary stage will also be notified). This will allow the community to share feedback on your project proposal, ask questions, and offer you suggestions.
- Allow 1 to 2 weeks after your project gets posted on the Oasis GitHub discussion board and receives community feedback for the Oasis team to mark your project as ACCEPTED or TURNED DOWN. We will also notify you if your application is approved or turned down at this stage via email.
- If your grant application is accepted, the Oasis team will email you with next steps, including information about the KYC process.
The Oasis team will work closely with you to finalize the terms of your grant, and you can begin working on your project!

-----

### HELPFUL DOCUMENTATION

The following documentation should help you as you’re writing your project proposal and thinking about how you’re going to build your app:
- Parcel SDK documentation http://docs.oasiscloud.io/ ( use this password: Parcel2020! )

For access to Parcel SDK please fill out this form:
- https://oasisfoundation.typeform.com/to/u3EINnMw

Check out the following tutorials and app demos:
- Parcel in Action: PrivGuard - https://youtu.be/SKYP2Vceag0
- Oasis Harvest Demo - https://youtu.be/9HduApRDJyA
- Parcel in Action: Nebula Genomics - https://youtu.be/VA5la2h75lQ
- Parcel in Action: MusicFund - https://youtu.be/DQqiiLYw1mI

For more ideas on the types of projects you can build using Parcel SDK, you can check out some of the projects that won The Quest for ROSE: An Oasis Network Hackathon: https://medium.com/oasis-protocol-project/announcing-the-winners-of-the-quest-for-rose-hackathon-5d160722a840

-----

### GET SUPPORT

If you need any help getting started, check out the following community support channels:
- Join our Oasis Community Slack to view all of our support channels for developers: https://join.slack.com/t/oasiscommunity/shared_invite/zt-8a71w8i9-sVhe_5IG1hhb5Dgjl5xu2A 
- For logistical questions related to the ROSE Bloom Grants Program, post in the #grants-program-public channel.
- For tech questions related to building with Parcel SDK, post in the #hackathon-parcel-support channel
- To learn more about the Oasis Network community of developers and supporters, join our Telegram: https://t.me/oasisprotocolcommunity 

-----

### TERMS & CONDITIONS

Please note that eligibility to get approved for a grant is subject to KYC and your country of residence's accredited investor verification guidelines, as outlined in the official program rules (https://docsend.com/view/xykrayt6vb5tn7eh). 

Developers who are U.S. citizens or residents and who are not accredited investors may not be eligible to receive ROSE Bloom Grants. The grants are not and will not be offered or sold, directly or indirectly, to any person who is a resident, organized, or located in any country or territory subject to OFAC comprehensive sanctions programs from time to time, including Cuba, Crimea region of Ukraine, Democratic People’s Republic of Korea, Iran, Syria, any person found on the OFAC Specially Designated Nationals, Blocked Persons List, any other consolidated prohibited persons list as determined by any applicable governmental authority, or any persons who are organized in, located in, or residents of New York State.

-----

### For other grant project categories, check out our other Grants Program discussion threads: https://github.com/oasisprotocol/community/discussions/categories/grants-program

-----

**Comment by jcueto** [comment]
### Grant Proposal: Integrating the Parcel API with our Dead Man's Switch
### Status: [Under Review][Gathering Community Feedback]

**What's your team background / professional profile?**

We have worked in Python, Scala, Rust, Javascript, and smart contract platforms including Solidity and Oasis's WASM infrastructure.  https://github.com/elsehow

**Please provide an overview of your project.**

Our previous project, funded by the Oasis Foundation, created a Dead Man's Switch using the Oasis platform [1]. (A Dead Man's switch releases information in the future if a user does not take a specified action in a specified timeframe).   This project fixes major problems with existing Dead Man's Switches by assuring there is no central point of failure for leaks or coercion. However, our existing product does not handle the encryption of large files. It works primarily as a key escrow, leaving the work of encrypting the files themselves---and of storing them on platforms such as IPFS---to the user. This introduces the potential for user error. Users may encrypt files incorrectly, or fail to encrypt them at all. They may place information in centralized repositories, where it can be made unavailable. They may also fail to link their Dead Man's Switch messages with the encrypted files they upload.  In this project, we propose integrating the Parcel API with our Dead Man's Switch, thus allowing the Oasis platform to handle all parts of the encryption process. The result will be a Dead Man's Switch that is not only distributed, but also resistant to user error. The result will be an idealized Dead Man's Switch, in which both the data and the key for decrypting it are kept confidential via the Oasis platform.  [1]. https://github.com/elsehow/nikkileaks

**What are the key milestones for your project?**

Milestone 1. Integrate Parcel API with existing Nikkileaks API. Milestone 2. Basic UX for posting leaks. Milestone 3. Complete user testing with users identified in prior Oasis Foundation grant.

**Comment by jcueto** [comment]
### Grant Proposal: Expanding Oasis-Docs application
### Status: [Under Review][Gathering Community Feedback]

**What's your team background / professional profile?**

I'm a full stack developer, with 10 years experience in games development (unity, cocos creator), AI (2 years from 2015 to mid 2017) Since 2017 I've been working as a full time blockchain developer.

**Please provide an overview of your project.**

Oasis-Docs, a privacy-first document system, which got Silver Prize in Oasis Network hackathon on Gitcoin.  At Oasis-Docs, users can store, edit, sharing their document without any leakage worries.

**What are the key milestones for your project?**

- Upload documents to Oasis function: completed at hackathon Next works are expected to be started from March: Phase 1: (1 month) - Managing document: sort/search/view - Sharing with others with specific permission Phase 2: (1 month) - Document editing/Removing - Collaboration editing - Files encryption Future: - Apply some AI models to document system, for example: Similarity model to detect similar documents, OCR

**Comment by jcueto** [comment]
### Grant Proposal: Private Key Custodianship
### Status: [Under Review][Gathering Community Feedback]

**What's your team background / professional profile?**

Our team has been working together since August 2019 when we started a project for the Ideo Colab hackathon and we ended up winning the Oasis track. That is when Lovecrypto was born!  Since then we continued working on the project and Lovecrypto is now an incorporated startup with a product, users, clients and some traction. To see the current state of Lovecrypto App, please see the Android Version (https://play.google.com/store/apps/details?id=br.com.lovecrypto.app) or the IOS version (https://apps.apple.com/us/app/lovecrypto/id1540132759).  Our team is made of:  Edmilson Rodrigues (CEO) - Worked at Google in 2012 and 2013. Currently is a MSC candidate in Computer Science at UFPE. At Lovecrypto is responsible for Project Management and Business Development. LinkedIN: https://www.linkedin.com/in/edmilsonrodrigues/ ; Github:  Ailson da Cruz (CTO and back end developer) - Former Projects Director at Citi (https://citi.org.br/). Undergraduate student in Computer Science at CIn/UFPE.  LinkedIN: https://www.linkedin.com/in/ailson-da-cruz-17b83b92/ ; Github: https://github.com/ailsoncgt  Lucas Zacarias (CPO and designer) - Previously Lucas was the co-founder of Guardaqui (https://www.linkedin.com/company/guardaqui). He is a talented designer and Product person. He is an undergraduate student in Computer Science at CIn/UFPE.  LinkedIN: https://www.linkedin.com/in/lucaszsd/ ; Github: https://github.com/lucaszsd  Juliandson Estanislau (Blockchain Expert) - Previously worked at FIEPE building an Ethereum blockchain project. At Lovecrypto, Juliandson is responsible for the Blockchain related development. Currently is a MSC candidate in Computer Science at UFPE. LinkedIN: https://www.linkedin.com/in/juliandson-ferreira-4218b8b4/ ; Github: https://github.com/stanis18

**Please provide an overview of your project.**

Cryptocurrency Wallet developers have to make a choice about being custodian of the users’ private keys or delegate to users that responsibility. When the wallets are non custodial, there is a trade off for usability because it is not easy for new users to deal with private keys or their 12-24 words mnemonic passphrases. That is why Lovecrypto decided to be custodial of the users’ private keys in its App. However, there is always the risk of our users private keys being compromised and a hacker to transfer their cryptocurrency. That is why Lovecrypto proposes the development of this project to use Oasis Network’s Paracel SDK and to create a new design for private keys custodianship that addresses 6 of the TOP 10 OWASP foundation Mobile Risks (https://owasp.org/www-project-mobile-top-10/):  M1: Improper Platform Usage M2: Insecure Data Storage M3: Insecure Communication M4: Insecure Authentication M5: Insufficient Cryptography M6: Insecure Authorization Our plan is to develop this project, integrate it into Lovecrypto App (see here the code: https://github.com/lucaszsd/LovecryptoAppScreenshots) and start using Paracel SDK to host our users’ private keys. Then, we will Open Source the project and try to create a community around it so that other Crypto Apps or Crypto exchanges that are custodian of the user’s private keys can use Oasis Network and its Paracel SDK for higher security.

**What are the key milestones for your project?**

#1 - Literature review and writing the Requirement Documents. Expected time to complete: 1 month Transfer upon Signature: $6,000 in Rose Tokens We are going to dive into the literature and best practices, reviewing the following papers:  https://dl.acm.org/doi/abs/10.1145/3318041.3355459  https://onlinelibrary.wiley.com/doi/abs/10.1111/1911-3838.12242  https://link.springer.com/chapter/10.1007/978-3-030-54455-3_19  https://ieeexplore.ieee.org/abstract/document/9169412  https://ieeexplore.ieee.org/abstract/document/9071475  https://dl.acm.org/doi/abs/10.1145/3357236.3395535  https://ieeexplore.ieee.org/abstract/document/8472760  https://era.library.ualberta.ca/items/9fab24c2-0007-4839-8c33-58a056d74052  https://ieeexplore.ieee.org/abstract/document/8686583   After that, we are going to write and submit the requirements document of the “HoldMyKeys” solution for your revision.    #2 - Develop HoldMyKeys and integrate it with Lovecrypto App’s backend. Expected time to complete: 3 months In this phase we are going to develop the infrastructure that addresses the top 6 risks in the OWASP Foundation, connect it to Paracel SDK, and implement it with Lovecrypto App.  Payment: $10,000 in ROSE tokens after submission. --- #3 - Open Sourcing the project. Involves the creation of a website explaining the project, creating and maintaining social media channels and creating a SDK to make it easily available for other developers or custodial Cryptocurrency Apps and Exchanges to implement “HoldMyKeys” in Paracel SDK.  Expected time: 6 month Payment: TBD Total time: 10 months

**Reply by edrodrigues** [reply]
Hi. I am from the development team proposing this project. We would lovet o create an open source community around it. Let me know if you have any doubts or would like to participate in this project once it is completed.
